index.src.html

```diff
@@ -782,25 +782,55 @@ <h4 id="algorithm-require-mediation" algorithm>Require User Mediation</h4>
             return;
           }
           if (credential.<a attribute for="Credential">type</a> == "<a const>password</a>") {
-            fetch("https://example.com/loginEndpoint", { credentials: credential, method: "POST" })
+            var form = new FormData();
+            form.append('username_field', credential.id);
+            form.append('password_field', credential.password);
+            fetch("https://example.com/loginEndpoint", { body: form, method: "POST" })
               .then(function (response) {
                 if (/* |response| indicates a successful login */) {
                   // Store the credential again, see note below.
                   navigator.<a attribute>credentials</a>.<a idl lt="store()" for="CredentialsContainer">store</a>(credential);
                   // Notify the user that sign-in succeeded! Do amazing, signed-in things!
+                  // Maybe navigate to a landing page via location.href =
+                  // '/signed-in-experience'?
                 } else {
                   // Insert some code here to fall back to a basic login form.
                 }
               });
           }
         });
     </pre>
+
+    Alternatively, the website could just copy the credential data into a <{form}> and call
+    {{HTMLFormElement/submit()}} on the form:
+
+    <pre>
+      navigator.<a attribute>credentials</a>
+        .<a idl lt="get()" for="CredentialsContainer">get</a>({ "<a for="CredentialRequestOptions" dict-member>password</a>": true })
+        .then(credential =&gt; {
+          if (!credential) {
+            return; // as above...
+          }
+          if (credential.<a attribute for="Credential">type</a> == "<a const>password</a>") {
+            document.getQuerySelector('input[name=username_field]').value =
+                credential.id;
+            document.getQuerySelector('input[name=password_field]').value =
+                credential.password;
+            document.getQuerySelector('#myform').submit();
+          }
+        });
+    </pre>
+
+    Note that the former method is much preferred, as it contains an explicit call
+    to {{CredentialsContainer/store()}} and saves the credentials. The <{form}> based mechanism
+    relies on form submission, which navigates the browsing context, making it difficult to
+    ensure that {{store()}} is called after successful sign-in.
```

Comment:
I have discussed this with @mikewest and I think this is where we ended up. Isn't the "Note that the former method is much preferred" strong enough?